### ISSUE TITLE
loss function in train() 

### ISSUE THREAD

**Issue by xht033** [CuriousCat-7/born-again-neural-network#1]
kl_divergence(output,ouput2) is not correct here.

**Comment by markaustinbrophy** [comment]
@xht033 it's supposed to be mse, correct? kl should work, just not quite as well!

**Comment by xht033** [comment]
first, kl_divergence is asymmetrical, and I think it should be kl_divergence(log_softmax(output), softmax(output2)).

**Comment by ShuxuanGuo** [comment]
I found @xht033 was right, just thinking whether this code could reproduce the results.

**Comment by GreatGBL** [comment]
I think the result should be 

loss2 = kldivergence(F.log_softmax(outputs, dim=0), F.softmax(outputs2, dim=0))
loss = loss + loss2.item()
